jar-submission-test/src/main/java/com/hazelcast/jet/tests/jarsubmission/JarSubmissionTest.java

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.tests.jarsubmission;
+
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.jet.JetService;
+import com.hazelcast.jet.Job;
+import com.hazelcast.jet.SubmitJobParameters;
+import com.hazelcast.jet.core.JobStatus;
+import com.hazelcast.jet.tests.common.AbstractSoakTest;
+import com.hazelcast.map.IMap;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static com.hazelcast.jet.tests.common.Util.sleepMillis;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+public class JarSubmissionTest extends AbstractSoakTest {
+
+    private static final int SMALL_JAR_LOG_JOB_COUNT_THRESHOLD = 100;
+    private static final int LARGE_JAR_LOG_JOB_COUNT_THRESHOLD = 50;
+
+    private static final int ASSERTION_ATTEMPS = 1200;
+    private static final int ASSERTION_SLEEP_MS = 100;
+
+    private static final String SMALL_JAR_PATH_DEFAULT = "/tmp/jar-submission-job.jar";
+    private static final String LARGE_JAR_PATH_DEFAULT = "/tmp/jar-submission-job-large.jar";
+
+    private static final int PAUSE_BETWEEN_SMALL_JAR_JOBS = 1_000;
+    private static final int PAUSE_BETWEEN_LARGE_JAR_JOBS = 3_000;
+    private static final int DELAY_AFTER_TEST_FINISHED = 120_000;
+
+    private String smallJarPath;
+    private String largeJarPath;
+    private int pauseBetweenSmallJarJobs;
+    private int pauseBetweenLargeJarJobs;
+
+    public static void main(String[] args) throws Exception {
+        new JarSubmissionTest().run(args);
+    }
+
+    @Override
+    protected void init(HazelcastInstance client) throws Exception {
+        smallJarPath = property("smallJarPath", SMALL_JAR_PATH_DEFAULT);
+        largeJarPath = property("largeJarPath", LARGE_JAR_PATH_DEFAULT);
```

Comment:
Let's copy the small jar here and add the large CSV file into it, then we can avoid the S3 hassle.

Reply:
Good idea, I'll improve that.